ui/src/components/Document/DocumentAuthors.js

```diff
@@ -1,28 +1,123 @@
-import { List } from 'semantic-ui-react';
+import { List, Popup, Icon } from 'semantic-ui-react';
 import React, { Component } from 'react';
 import PropTypes from 'prop-types';
+import isEmpty from 'lodash/isEmpty';
 
 export class DocumentAuthors extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      expanded: false,
+    };
+  }
+
+  renderPopupContent = author => {
+    return (
+      <>
+        {author.full_name}
+        <br />
+        {!isEmpty(author.roles) && (
+          <>
+            <span>{author.roles.toString()}</span>
+            <br />
+          </>
+        )}
+        {!isEmpty(author.affiliations) && (
+          <>
+            <span>
+              Aff. :{' '}
+              {author.affiliations
+                .map(item => {
+                  return item.name;
+                })
+                .toString()}
+            </span>
+          </>
+        )}
+      </>
+    );
+  };
+
+  renderPopup = author => {
+    return (
+      <>
+        {' '}
+        <Popup
+          content={this.renderPopupContent(author)}
+          position="top center"
+          flowing
+          trigger={<Icon name="question circle outline" />}
+        />
+      </>
+    );
+  };
+
+  toggleExpandedState = () => {
+    this.setState({ expanded: !this.state.expanded });
+  };
+
+  renderExpandButton = () => {
+    return (
+      <>
+        {' '}
+        <span className="button-show-more" onClick={this.toggleExpandedState}>
+          {this.state.expanded ? 'Show less' : 'et al.'}
+        </span>
+      </>
+    );
+  };
+
   render() {
-    const { metadata, prefix, otherAuthorsDisplay, delimiter } = this.props;
+    const {
+      metadata,
+      prefix,
+      otherAuthorsDisplay,
+      popupDisplay,
+      delimiter,
+      authorsLimit,
+    } = this.props;
     const otherAuthors = otherAuthorsDisplay ? otherAuthorsDisplay : 'et al.';
+
+    let authors = metadata.authors;
+    let classes = 'document-authors-list-wrapper';
```

Comment:
to improve readability you could just add "expanded" to the class, f.e (pseudocode)
```
const expandedClass = this.state.expanded && "expanded";
```

and then below, in the div:

```
`document-authors-list-wrapper ${expandedClass}`
```